[Web] 비회원 주문서 입력 폼 기능 개선 #121

MARUNGs opened the pull request:
💼 작업 설명
비회원 주문서 입력 폼 기능 개선

✅ 주문자, 수령인 정보가 동일한 경우 체크박스를 체크하게하고, 체크박스 값이 true이면 수령인 Input disabled 처리
✅ 주문 비밀번호 placeholder 수정
✅ 개인정보 수집 및 이용 동의서 모달 패딩값 필요, 백그라운드 딤 처리, 닫기 버튼 고정
✅ 예금주 특수문자 허용 (, ), A-Z, _, 가-힣
✅ 틀린 값을 입력한 뒤 submit 버튼 클릭 시 오류가 발생하고 다시 정상으로 입력하면 '주문자' 오류메세지는 사라지고 '수령인' 오류메세지는 그대로 남는 이슈 확인
✅ 배송방법 - deliveryType === '10'이 아닌 option 선택 시 배송 메세지 disabled 처리
✅ 배송방법 select 값 변경 시 배송메세지 초기화
✅ 배송비 추가 :: 배송방법 - fee 이용 (deliveryType === '10' 한정)
✅ api 호출 위치 변경 :: 상위 컴포넌트에서 하위 컴포넌트로 prop 전달하는 방식으로 변경하기
✅ 주문 전 확인메세지 alert confirm 기능 추가

MARUNGs committed:
refactor(web): 비회원 주문서 입력 폼 기능 개선 #121

MARUNGs committed:
refactor(web): 배송비 추가

* 배송방법이 '우체국'이면 배송비 3,000원 추가되고 최종금액에 반영되도록 처리
* API 호출 위치 변경 : 하위 컴포넌트에서 호출된 API를 상위 컴포넌트로 옮김처리하였음

MARUNGs committed:
refactor(web): 주문 전 안내메세지 추가

silo9506 reviewed (approved):
체크된 상태로 주문자 수정하면 수령인은 변경이 안됨 
체크하면 입력요소를 없애버리고 서브밋할때 자동으로 값 넘겨주는식으로 하는건 어떨까 싶음
<img width="634" height="175" alt="image" src="https://github.com/user-attachments/assets/7eb4b344-ea30-497f-bcf0-184eadb6e706" />

MARUNGs committed:
refactor(web): 추가 수정사항 #121

✅ 주문자-수령인 체크박스 true인 상태에서 주문자 수정 시 수령인 정보 변경 안 되는 이슈 :: 주문자 수정 시 체크박스 false, 수령인 정보 disabled=false로 처리되도록 변경하기.
✅ 주문하기 버튼 :: 기능 원복하기
✅ 배송지 :: 결과값의 상세주소는 빈값으로 처리
✅ 배송지 :: 주소검색 이전의 상세주소 disabled 처리
✅ 모달 잘림 현상 :: 최대높이 600제한, 나머지는 스크롤 영역 처리

MARUNGs commented:
> 체크된 상태로 주문자 수정하면 수령인은 변경이 안됨 체크하면 입력요소를 없애버리고 서브밋할때 자동으로 값 넘겨주는식으로 하는건 어떨까 싶음 <img alt="image" width="634" height="175" src="https://private-user-images.githubusercontent.com/110013101/468469413-7eb4b344-ea30-497f-bcf0-184eadb6e706.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMwODAzMzMsIm5iZiI6MTc1MzA4MDAzMywicGF0aCI6Ii8xMTAwMTMxMDEvNDY4NDY5NDEzLTdlYjRiMzQ0LWVhMzAtNDk3Zi1iY2YwLTE4NGVhZGI2ZTcwNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyMVQwNjQwMzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wM2Y1ODMzZjgxM2EyYjg0MTE1MjFkYjg1MWQzMWM0ZTY2MjViODc0OWEzYTU1M2I1YzgyYTg5YzRlODc4MjQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.OalBYzuhJNsUlRr1O57QKTa9WqkonYxLji7bCaXkV4M">




수령인은 사용자에게 정보를 제공해줘야 하는 필드인데 그게 사라지면 애매해질 것 같숨
